ts/ts-bas/src/test/java/se/inera/intyg/common/ts_bas/v7/model/converter/certificate/question/QuestionIntygetAvserTest.java

```diff
@@ -0,0 +1,304 @@
+                        .label(texts.get(INTYG_AVSER_IAV10_LABEL_ID))
+                        .build());
+            }
+
+            @Override
+            protected Integer getRows() {
+                return 2;
+            }
+
+            @Override
+            protected Integer getColumns() {
+                return 0;
+            }
+        }
+
+        @Nested
+        class IncludeValueCheckboxMultipleCodeTest extends ValueCheckboxMultipleCodeTest {
+
+
+            @Override
+            protected CertificateDataElement getElement() {
+                return QuestionIntygetAvser.toCertificate(null, 0, texts);
+            }
+
+            @Override
+            protected CertificateDataElement getElementWithValues() {
+                return QuestionIntygetAvser.toCertificate(Set.of(
+                    IntygAvserKategori.IAV3, IntygAvserKategori.IAV2, IntygAvserKategori.IAV1), 0, texts);
+            }
+
+            @Override
+            protected List<Object> getValues() {
+                return List.of(
+                    CertificateDataValueCode.builder()
+                        .id(IntygAvserKategori.IAV1.name())
+                        .code(IntygAvserKategori.IAV1.name())
+                        .build(),
+                    CertificateDataValueCode.builder()
+                        .id(IntygAvserKategori.IAV2.name())
+                        .code(IntygAvserKategori.IAV2.name())
+                        .build(),
+                    CertificateDataValueCode.builder()
+                        .id(IntygAvserKategori.IAV3.name())
+                        .code(IntygAvserKategori.IAV3.name())
+                        .build()
+                );
+            }
+        }
+
+        @Nested
+        class IncludeValidationMandatoryTest extends ValidationMandatoryTest {
+
+            @Override
+            protected String getQuestionId() {
+                return INTYG_AVSER_SVAR_ID_1;
+            }
+
+            @Override
+            protected String getExpression() {
+                return IntygAvserKategori.IAV1.name() + " || " + IntygAvserKategori.IAV2.name() + " || "
+                    + IntygAvserKategori.IAV3.name() + " || " + IntygAvserKategori.IAV4.name() + " || "
+                    + IntygAvserKategori.IAV5.name() + " || " + IntygAvserKategori.IAV6.name() + " || "
+                    + IntygAvserKategori.IAV7.name() + " || " + IntygAvserKategori.IAV8.name() + " || "
+                    + IntygAvserKategori.IAV9.name() + " || " + IntygAvserKategori.IAV10.name();
+            }
+
+            @Override
+            protected CertificateDataElement getElement() {
+                return QuestionIntygetAvser.toCertificate(null, 0, texts);
+            }
+
+            @Override
+            protected int getValidationIndex() {
+                return 0;
+            }
+        }
+    }
+
+    @Nested
+    class ToInternal {
+
+        @Test
+        void shouldIncludeKorkortsTyp() {
```

Comment:
Any possibility to create a textFixture for managing CodeList? The challenge is similar as for the Value-text-fixture.

Reply:
Resolved through generic testFixtures.